api/internal/core/store/store.go

```diff
@@ -109,12 +110,20 @@ func ReInit() error {
 			return err
 		}
 	}
+
+	//clear storeNeedReInit
+	storeNeedReInit = storeNeedReInit[:0]
```

Comment:
I think only add this line is enough.

Reply:
when for loop return err, this line will not be excute,and the same store may be add to the storeNeedReInit slice multiply times, and in some condition the storeNeedReInit slice may increase to a very huge size

Reply:
![image](https://github.com/apache/apisix-dashboard/assets/5869520/f119ab58-2a23-4309-9c26-4ac15eff1e58)
as this pic show, manager-api process memory increase to 21GB!

Reply:
but if return err ,  it can't reach code  
`storeNeedReInit = append(storeNeedReInit, s) `

Reply:
so i add some other protect:

![image](https://github.com/apache/apisix-dashboard/assets/5869520/23051967-2005-470e-b302-1be3d4d034a0)

Reply:
only add the store to the slice when it's inited changed from true to false

Reply:
so  if return err , it can't reach code  ,   why multi  times ?

Reply:
@hanqingwu do you accept this pr?

Reply:
yes, I think this PR works .